server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -513,8 +537,94 @@ public ClusterMetadataManifest getLatestClusterMetadataManifest(String clusterNa
         return fetchRemoteClusterMetadataManifest(clusterName, clusterUUID, latestManifestFileName);
     }
 
+    /**
+     * Fetch the previous cluster UUIDs from remote state store and return the most recent valid cluster UUID
+     *
+     * @param clusterName The cluster name for which previous cluster UUID is to be fetched
+     * @return Last valid cluster UUID
+     */
+    public String getLatestClusterUUID(String clusterName) {
+        try {
+            Set<String> clusterUUIDs = getAllClusterUUIDs(clusterName);
```

Comment:
This contains clusterUUID of the current cluster as well, right?

Reply:
Correct